src/providers/common.ts

```diff
@@ -238,3 +243,20 @@ export interface IRunnerProvider extends ec2.IConnectable, iam.IGrantable {
    */
   getStepFunctionTask(parameters: RunnerRuntimeParameters): stepfunctions.IChainable;
 }
+
+/**
+ * Base class for all providers with common methods used by all providers.
+ *
+ * @internal
+ */
+export abstract class BaseProvider extends Construct {
+  protected labelsFromProperties(defaultLabel: string, propsLabel: string | undefined, propsLabels: string[] | undefined): string[] {
```

Comment:
I would prefer optional parameters over explicit union types:
```suggestion
  protected labelsFromProperties(defaultLabel: string, propsLabel?: string, propsLabels?: string[]): string[] {
```

Reply:
That would allow implementing classes to not pass anything for `propsLabels`. I had a bug while updating the code where I forgot just one of the providers with:

```
this.labels = this.labelsFromProperties('codebuild', props.label);
```

It compiled just fine. I would rather this be a compile error.